docs/node/statesync.mdx

```diff
@@ -0,0 +1,80 @@
+
+```toml
+[app.snapshots]
+
+# Enables snapshots
+enabled = true
+
+# Path to the snapshots directory
+snapshot_dir = "snapshots"
+
+# (Multiples of) Heights at which the database is snapshotted.
+recurring_height = 10000
+
+# Maximum number of snapshots to store on disk
+max_snapshots = 3
+```
+
+If snapshots are enabled, node starts to generate snapshots at every block height that is a multiple of `recurring_height`. The snapshots generated are stored on the disk at the location specified using the `snapshot_dir` configuration. These snapshots are the logical sql dumps of the underlying Postgres database, compressed and divided into chunks of 16MB each. 
+
+
+## Configuring Statesync
+
+The new node joining the network can be configured to use statesync by providing the following configuration in the `config.toml` file.
+
+```yaml
+[chain.statesync]
+# State sync rapidly bootstraps a new node by discovering, fetching, and restoring the database
+# snapshot from peers instead of fetching and replaying historical blocks. Requires some peers in
+# the network to take and serve snapshots. State sync is not attempted if the node
+# has any local state (LastBlockHeight > 0). The node will have a truncated block history,
+# starting from the height of the snapshot.
+enable = true
+
+# Path to the directory where the received snapshot is stored
+snapshot_dir = "rcvd_snaps"
+
+# Trusted snapshot providers (comma-separated chain RPC servers) are the source-of-truth for the snapshot integrity.
+# Snapshots are accepted for statesync only after verifying it with these trusted snapshot providers.
+# Atleast 1 trusted rpc server is required for enabling state sync.
+rpc_servers = "http://rpc1:26657,http://rpc2:26657"
+
+# Time to spend discovering snapshots before initiating a restore. (default: 15 seconds)
+discovery_time = "15s"
+
+# The timeout duration before re-requesting a chunk, possibly from a different
+# peer (default: 1 minute), if the current peer is unresponsive to the chunk request.
+chunk_request_timeout = "10s"
+
+# Note: If the requested chunk is not received within 2 minutes (hard-coded default), 
+# the state sync process is aborted and the node will fall back to the regular block sync process.
+```
+
+When state sync is enabled, the node first discovers snapshots from all its connected peers. It then selects the latest snapshot from those discovered and validates the integrity of the snapshot with the trusted snapshot provider. Once a valid snapshot is identified, the node fetches the snapshot chunks from the peers and restores the database state using these chunks. The node then begins syncing blocks starting from the snapshot height.
+
+:::note
+The node will stay in the discovery phase until a snapshot is discovered and validated. If there are no snapshots in the network, the node will be stuck in the discovery phase. To make progess, restart the node by disabling statesync to switch to using blocksync.
```

Comment:
```suggestion
The node will stay in the discovery phase until a snapshot is discovered and validated. If there are no snapshots in the network, the node will be stuck in the discovery phase. To make progess, disable statesync and restart the node. The node will then progress with blocksync.
```

Reply:
Are there any docs we can hyperlink to on blocksync?

Reply:
Umm, I can add one. But there is not much to talk about it. I will probably add one at the top of this file.